### PR TITLE
Add docs and tests for utils

### LINKED ISSUE #43: Utils/camelizeKeys

### LINKED ISSUE #46: Utils/hasOverlap

### LINKED ISSUE #47: Utils/decamelizeKeys

### LINKED ISSUE #48: Utils/omitUndefined

### PR DESCRIPTION
**Pull request by dpikt**
Resolves #43 
Resolves #46 
Resolves #47 
Resolves #48 

Contains the same annoying docs changes as #62...

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -14,21 +14,26 @@ export { default as union } from 'lodash/union'
...
 export { onLoad } from '@launchpadlab/lp-utils'
 
+// Returns true when two arrays share common items
 export function hasOverlap (arr1, arr2) {
   const overlapItems = overlap(arr1, arr2)
   return (overlapItems.length > 0)
 }
 
+// Converts all keys of the given object to camelCase
 export function camelizeKeys (obj) {
```
Comment (davemcorwin) [inline src/utils.js]: the docs should mention that this is wrapper around `humps` and the docs and test should address the exceptional case when a key starts with a `_`, it is not transformed.

File: src/utils.js [hunk 2]
```diff
@@ -14,21 +14,26 @@ export { default as union } from 'lodash/union'
...
   return humps.camelizeKeys(obj, (key, convert) =>
     /^_/.test(key) ? key : convert(key)
   )
 }
 
+// Converts all keys of the given object to lower_case
```
Comment (davemcorwin) [inline src/utils.js]: the docs should mention that this is wrapper around `humps`

File: src/utils.js (hunk 1 above)
Comment (dpikt) [inline src/utils.js]: 👍  good call

**Comment by dpikt** [comment]
@davemcorwin updates made

**Comment by davemcorwin** [comment]
@dpikt hmmm, did you update the comments?

**Comment by dpikt** [comment]
@davemcorwin I didn't save them.......... :(

**Comment by dpikt** [comment]
@davemcorwin here you go

**Comment by dpikt** [comment]
Merged into #64

**Pull request closed without merging**